build.go

```diff
@@ -123,32 +122,31 @@ func Build(build BuildFunc, options ...Option) {
 		ok   bool
 	)
 	ctx := BuildContext{}
-	logger := log.New(os.Stdout)
```

Comment:
Do we want to set a default logger in config at L103?

Reply:
I avoided this because I didn't want anything to consume libcnb's `log` package by default. If we add a default, then either config or main references the `log` package. 

I know that Go can do some smart things around stripping out code that is unused to reduce binary size. I don't know Go well enough to know exactly what it can do though. My fear was that by including a default, we'd be forcing Go to always compile in the code for this default log implementation (what in the `log` package), even if a buildpack author provides their own logging implementation. That fear may be unfounded, maybe Go is smart enough to still strip it out but I didn't know how to test it either, so I just left it out.

Right now, the logger gets passed into the `func Main(...)` method as an `Option` and then set on the `Config` object, but that makes it optional. Maybe the way to go here is to make it a required argument to `func Main(...)` so that it's never nil. 

Thoughts?

Reply:
I think just having some check to make sure that the default is not nil would be nice. I would personally prefer if we had a sane default of "plainlogger" but I can understand the reasoning behind not wanting to import that.

Reply:
I can certainly add a nil check. It's not the nicest experience though because I basically have to fail and say there's no logger.

Tbh, I don't have really strong feelings about the whole import situation. I had heard that come up before as a concern and so I thought I would try to keep that into consideration with the design. I have a gut feeling that it's probably a premature optimization given the size of the log package, which is very small, but again, I don't really know how to measure/test/prove that either besides just compiling with and w/out the import and checking the difference.

If you feel like the ergonomics/usage of the API is better by having a default, I'd be OK with that option too.

Reply:
I was playing around with [this tool](https://github.com/knz/go-binsize-viz) for examining the size of a buildpack binary. I looked at a couple of the main binaries for Paketo buildpacks. 

Here are some stats from one:

```
19% runtime/
16% github.com/
15% crypto/
13% net/
8.4% vendor/golang.org/x
```

Then a bunch of stuff in the 3.5% and lower range. Things like `encoding/`, `regexp/`, `math/`, `time/`, `os/`, etc...

Within `github.com/`

```
3.9% paketo-buildpacks/
3.9% pelletier/go-toml
3.5% BurntSushi/toml
1.2% or 40k buildpacks/libcnb
1.2% h2non/filetype
```

Within `buildpacks/libcnb` 0.044% or 1.5k is poet.

I guess based on that, it's probably not really critical to separate these packages. The additional overhead of a few small classes is just going to be insignificant. I'll go ahead and make that change.

Reply:
Default added.